src/app/qgsmaptoolselectutils.cpp

```diff
@@ -335,3 +337,193 @@ QgsFeatureIds QgsMapToolSelectUtils::getMatchingFeatures( QgsMapCanvas *canvas,
   return newSelectedFeatures;
 }
 
+
+QgsMapToolSelectUtils::QgsMapToolSelectMenuActions::QgsMapToolSelectMenuActions(
+  QgsMapCanvas *canvas,
+  QgsVectorLayer *vectorLayer,
+  QgsVectorLayer::SelectBehavior behavior,
+  QObject *parent ):
+  QObject( parent ),
+  mCanvas( canvas ),
+  mVectorLayer( vectorLayer ),
+  mBehavior( behavior )
+{
+  connect( mVectorLayer, &QgsMapLayer::destroyed, this, &QgsMapToolSelectMenuActions::onLayerDestroyed );
+}
+
+QList<QAction *> QgsMapToolSelectUtils::QgsMapToolSelectMenuActions::actions( const QgsFeatureIds &featureCanditateIds )
+{
+  QList<QAction *> actionsList;
+
+  QString beginningOfText;
+
+  QgsFeatureIds effectiveFeatureIds = featureCanditateIds;
+
+  switch ( mBehavior )
+  {
+    case QgsVectorLayer::SetSelection:
+      beginningOfText = tr( "Select " );
```

Comment:
This is very tricky to translate (changes are high that the translated text will be missing the space "RemoveFeature One" or that some languages want to change the word order), better provide complete template texts "Select %1".